_posts/2017-11-07-haskell-learning-group-our-progress-and-updates.md

```diff
@@ -0,0 +1,71 @@
+
+One thing we've noticed and that was to expect, is that the difficulty of a content is slowly increasing. We are spending more time reading chapters. Sometimes the same material has to be read multiple times in order to fully understand what is it about.
+
+It helps a lot that the most of the concepts are familiar to us through other languages or libraries we are using. Let's say JavaScript doesn't support currying natively the way Haskell does it, but there are libraries which introduce these concepts. We are very often using [RamdaJS][ramdajs] in our projects, so such things as function composition, currying, partial application and so on were not something totally new to us. They were still had to be "translated" and demonstrated in a Haskell way but we certainly didn't need much time to get the idea.
+
+Similar situation was with typing system and type definitions. As many people from the group doing i.e. TypeScript or Scala, there were not that many surprises for us. Nevertheless it's pretty impressive what can be achieved with type constraints in Haskell.
+
+[ramdajs]: http://ramdajs.com
+
+## Types
+
+We went through many topics in the book and since the last update we've covered chapter four, five and discussed a half of the chapter six which covers Typeclasses. The main topics were:
+* Basic DataTypes
+* Types
+* Typeclasses
+
+We learned important basics that helps a lot when you read and try to understand what certain code does. For instance understanding how to read type signature, so something like this starting to make sense:
+
+`(Num a, Num b) => a -> b -> b`
+
+In this case there are two constraints, one for each parameter. They must be of a typeclass Num. You might already have some idea what function might do, based on its type signature.
+We had a look on different examples of how type signatures limit a scope of what body of the function might be. Even more, some  type signatures of function tells us exactly what it does because its the only possible implementation of the function. The simplest example of that will be an identity function:
+
+```haskell
+id :: a -> a
+id x = x
+```
+
+The only thing this function can possibly do is return parameter that was passed to it. More advanced example when there is only one possible implementation is function composition:
+
+```haskell
+f :: (b -> c) -> (a -> b) -> (a -> c)
+f x y = \z -> x(y(z))
+```
+
+It's not obvious from the first look at this type definition that there is no other possible implementation of a function based on its type. Compiler won't let us implement it in any other way. 
+
+## Katas and Exercises
```

Comment:
as much as I love this advice and your experience here - Maybe move it to new article? :) It would be very precious as a source of additional material for learning Haskell. Don't know, though, if this article is a good place for it.

But again - I would love to have it in separate article.

Reply:
idk, IMO this paragraph on its own is not worth to be a blogpost. The other option would be to remove it completely, but I see no reason why it doesn't belong here.

Reply:
It's just... like out of the context a bit - By all means - I like the idea, the advice you give here - I'm just afraid it won't stick to `group progress and updates`... Certainly it would be to short to call it `new article` - how about a little COOP so both of us could thing about making it bigger and more suitable as end-to-end guide for Haskell beginner?:)

Reply:
I like it.

Reply:
@KamWis I see your point, but following this principle, we might argue if Meetup section  belongs to this blogpost as well. I would prefer to keep it

Reply:
ok :)